bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewHubHandler.java

```diff
@@ -143,6 +147,21 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private synchronized void pollShadesAsync() {
+        pollShadesOnceFuture = scheduler.submit(() -> {
```

Comment:
No need to check if one is still scheduled ?

Reply:
Good point!

I'm in doubt about best approach here. Was reacting on @andrewfg's comment, since probably it's a bad idea to block the handler from `handleCommand`? But actually have mixed feelings about it. We already have blocking calls in same method for activating scenes etc. from same method, so it's already inconsistent. Only polling job is scheduled. Do you know if it's not allowed to have blocking network calls in `handleCommand`? It would certainly simplify the implementation if it wouldn't be needed.

It's fire-and-forget operations, so only keeping the reference for being cancelled correctly in dispose(), but now only cancels the last operation if multiple scenes are triggered very quickly as you've found.

Reply:
It depends what you mean by "blocking" :)
I think it is not allowed to have a really blocking call in `handleCommand`.
But you can run a HTTP request for example, without the need to create a separate thread.

Reply:
@lolodomo - I meant blocking as in I/O (network) operation, rendering thread blocked for possibly several hundreds milliseconds. :) Okay, if this is common, I'll revert commit d99aa701395ff28e29c259ee0ff01c45094c3fac to simplify the code and avoid unwanted behaviour.

Reply:
Reverted.

Reply:
@lolodomo - with @andrewfg found a better solution (see other comments). Ready again.